streaming/src/channel.cc

```diff
@@ -0,0 +1,277 @@
+
+  std::vector<ObjectID> queue_ids, failed_queues;
+  queue_ids.push_back(channel_info.channel_id);
+  upstream_handler->WaitQueues(queue_ids, 10 * 1000, failed_queues);
+
+  STREAMING_LOG(INFO) << "q id => " << channel_info.channel_id << ", queue size => "
+                      << channel_info.queue_size;
+
+  return StreamingStatus::OK;
+}
+
+StreamingStatus StreamingQueueProducer::DestroyTransferChannel() {
+  return StreamingStatus::OK;
+}
+
+StreamingStatus StreamingQueueProducer::ClearTransferCheckpoint(
+    uint64_t checkpoint_id, uint64_t checkpoint_offset) {
+  return StreamingStatus::OK;
+}
+
+StreamingStatus StreamingQueueProducer::NotifyChannelConsumed(uint64_t channel_offset) {
+  queue_->SetQueueEvictionLimit(channel_offset);
+  return StreamingStatus::OK;
+}
+
+StreamingStatus StreamingQueueProducer::ProduceItemToChannel(uint8_t *data,
+                                                             uint32_t data_size) {
+  Status status =
+      PushQueueItem(channel_info.current_seq_id + 1, data, data_size, current_time_ms());
+
+  if (status.code() != StatusCode::OK) {
+    STREAMING_LOG(DEBUG) << channel_info.channel_id << " => Queue is full"
+                         << " meesage => " << status.message();
+
+    // Assume that only status OutOfMemory and OK are acceptable.
+    // OutOfMemory means queue is full at that moment.
+    STREAMING_CHECK(status.code() == StatusCode::OutOfMemory)
+        << "status => " << status.message()
+        << ", perhaps data block is so large that it can't be stored in"
+        << ", data block size => " << data_size;
+
+    return StreamingStatus::FullChannel;
+  }
+  return StreamingStatus::OK;
+}
+
+Status StreamingQueueProducer::PushQueueItem(uint64_t seq_id, uint8_t *data,
+                                             uint32_t data_size, uint64_t timestamp) {
+  STREAMING_LOG(INFO) << "StreamingQueueProducer::PushQueueItem:"
+                      << " qid: " << channel_info.channel_id << " seq_id: " << seq_id
+                      << " data_size: " << data_size;
+  Status status = queue_->Push(seq_id, data, data_size, timestamp, false);
+  if (status.IsOutOfMemory()) {
+    status = queue_->TryEvictItems();
+    if (!status.ok()) {
+      STREAMING_LOG(INFO) << "Evict fail.";
+      return status;
+    }
+
+    status = queue_->Push(seq_id, data, data_size, timestamp, false);
+  }
+
+  queue_->Send();
+  return status;
+}
+
+StreamingQueueConsumer::StreamingQueueConsumer(std::shared_ptr<Config> &transfer_config,
+                                               ConsumerChannelInfo &c_channel_info)
+    : ConsumerChannel(transfer_config, c_channel_info) {
+  STREAMING_LOG(INFO) << "Consumer Init";
+}
+
+StreamingQueueConsumer::~StreamingQueueConsumer() {
+  STREAMING_LOG(INFO) << "Consumer Destroy";
+}
+
+StreamingStatus StreamingQueueConsumer::CreateTransferChannel() {
+  // subscribe next seq id from checkpoint id
+  // pull remote queue to local store if scheduler connection is set
```

Comment:
// Subscribe next seq id from checkpoint id and pull remote queue to local store if scheduler connection is set.

Reply:
> // Subscribe next seq id from checkpoint id and pull remote queue to local store if scheduler connection is set.

There is no subscription in the streaming queue, and pulling data has not been supported, so remove this comment.